src/ceph-disk/ceph_disk/main.py

```diff
@@ -1010,30 +1025,44 @@ def allocate_osd_id(
     cluster,
     fsid,
     keyring,
+    path,
 ):
-    """
-    Accocates an OSD id on the given cluster.
-
-    :raises: Error if the call to allocate the OSD id fails.
-    :return: The allocated OSD id.
-    """
+    lockbox_path = os.path.join(STATEDIR, 'osd-lockbox', fsid)
+    lockbox_osd_id = read_one_line(lockbox_path, 'whoami')
+    osd_keyring = os.path.join(path, 'keyring')
+    if lockbox_osd_id:
+        LOG.debug('Getting OSD id from Lockbox...')
+        osd_id = lockbox_osd_id
+        shutil.move(os.path.join(lockbox_path, 'osd_keyring'),
+                    osd_keyring)
+        path_set_context(osd_keyring)
+        os.unlink(os.path.join(lockbox_path, 'whoami'))
+        return osd_id
 
     LOG.debug('Allocating OSD id...')
+    secrets = Secrets()
     try:
-        osd_id = _check_output(
-            args=[
+        wanttobe = read_one_line(path, 'wanttobe')
+        if os.path.exists(os.path.join(path, 'wanttobe')):
+            os.unlink(os.path.join(path, 'wanttobe'))
+        id_arg = wanttobe and [wanttobe] or []
+        osd_id = command_with_stdin(
+            [
                 'ceph',
                 '--cluster', cluster,
                 '--name', 'client.bootstrap-osd',
                 '--keyring', keyring,
-                'osd', 'create', '--concise',
+                '-i', '/dev/fd/0',
```

Comment:
@dachary how do you think it is not necessary? Could you expand a bit on that and how this would increase the *"attack surface"* ? This implementation is trying to work around the caveats of requiring a file, but as noted in other comments, we might not want to write to a file.

So which one is it? Are we OK writing to a file? or we aren't because we might use sensitive information? `ceph-disk` might implement this workaround, but my concern is other tools as well. 

/cc @leseb @andrewschoen  @guits